app/helpers/order_helper.rb

```diff
@@ -26,13 +35,19 @@ def calculate_discount(line_items)
       end
     end.compact
 
+    # Could solve with a single iteration of the sum call:
+    #
+    # items[:line_items].sum { |item| item[:discounted_price] }
```

Comment:
I love this way, but Is the other way more complex to code?

Reply:
Not really, you would avoid a `pluck` and then `sum` operation and could send use a block to solve the sum issue in a single iteration:

```ruby
total_discount = items[:line_items].sum { |item| item[:discounted_price] }
```